### DISCUSSION TITLE
Header won't load if in other file than main branch and how to change url

### DISCUSSION THREAD

**Discussion by unviere** [community/community#131305] [Programming Help]
### Select Topic Area

Question

### Body

So I load the header in each file but it only load correctly in the main  but not in other files like settings folder how can I load it correctly in other files i will like make multiple pages but also sub pages and how can I remove.html from the link 

![Screenshot_20240707_103456_Spck Editor](https://github.com/community/community/assets/145258065/f2efae37-42da-4056-8841-e3988c8f187a)
![Screenshot_20240707_103049_Chrome](https://github.com/community/community/assets/145258065/b50e2a10-58fb-479a-8eae-ddd1b2fe257a)
![Screenshot_20240707_103056_Chrome](https://github.com/community/community/assets/145258065/a528711e-646e-4c42-98d4-0f50581b6e24)

```
   <div id="header"></div>
   
   <div>lol hi</div>
   
    <script src = "https://foodeggs7.github.io/Unviere/Scripts/topbar.js"></script>
```

I tried it with../Scripts/topbar.js and Scripts.topbar.js but all they return 403 forbidden but there should be a way to load it right I need to make sub pages like events/name of event I think../ isn't allowed but another method idk

**Comment by paiineli** [comment]
Hi,

It sounds like you're having issues loading your header across different pages and handling URLs. Here are some suggestions to help you out:

Loading the Header: Ensure that your topbar.js script correctly selects and inserts the header into each page. You might want to check that the script is properly targeting the #header div. If you’re getting a 403 Forbidden error, double-check that the file path is correct and that the server allows access to those files.

Example of how to load the header:

fetch('https://foodeggs7.github.io/Unviere/Scripts/header.html')
  .then(response => response.text())
  .then(data => {
    document.getElementById('header').innerHTML = data;
  })
  .catch(error => console.error('Error loading header:', error));

Removing .html from Links: To remove .html from your URLs, you can use a server-side solution or configure your project to handle routing with JavaScript. If you're using GitHub Pages, consider using a static site generator (like Jekyll) or a framework like React or Vue.js that can handle routing more effectively.

Folder Structure: Make sure your file paths are correct relative to your folder structure. If your settings folder is nested, the path might need to reflect that. For example:

<script src="../Scripts/topbar.js"></script>

Subpages and Routing: For creating subpages, consider using hash-based routing or a front-end framework that manages routes without relying on the server.

If you continue to encounter issues, sharing your folder structure could help pinpoint the problem further. Good luck!

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @unviere ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by unviere** [comment]
@paiineli it worked using the link and for the link if I use the link what can without.html I can acces the page but not on the unpublished local host but how can I do it easier with the jykel or something may the link change

**Comment by tienanh109** [comment]


**Reply by LiteBrite82** [reply]
Hi @tienanh109 , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Reply by tienanh109** [reply]
oops, ok srry
